### PR TITLE
Add support for Buffer secrets (#214)

### LINKED ISSUE #214: Secret should accept a Buffer (or array of buffers)
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

The type validation for `secret` in the plugin config and the signer implementation should be modified to allow `secret` to also be a `Buffer` instead of a `string`. This would mean modifying the checks for `typeof secret` to be expanded to also check `Buffer.isBuffer`.

### Motivation

The signer implementation already uses `crypto.randomBytes`, which returns a `Buffer`, to validate the algorithm. When generating secrets dynamically it would be useful to be allowed to simply pass the `Buffer` directly instead of having to convert it to a string representation like hex or base64 first.

### Example

```js
const secret = crypto.randomBytes(32); // new secret on every restart
app.register(require("@fastify/cookie"), { secret });
```

### PR DESCRIPTION
**Pull request by pluma**
Fixes #214.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: signer.js [hunk 1]
```diff
@@ -25,8 +25,8 @@ function Signer (secrets, algorithm = 'sha256') {
 
 function validateSecrets (secrets) {
   for (const secret of secrets) {
-    if (typeof secret !== 'string') {
-      throw new TypeError('Secret key must be a string.')
+    if (typeof secret !== 'string' && !Buffer.isBuffer(secret)) {
```
Comment (jsumners) [inline signer.js:28]:
```suggestion
    if (typeof secret !== 'string' && Buffer.isBuffer(secret) === false) {
```

**Review by Uzlopak — CHANGES_REQUESTED** [review]
I miss a test if providing a buffer actually gets the right result.

Here https://github.com/fastify/fastify-cookie/blob/e1394d0186bb86d62490b7632e0698115d4cf1c1/test/signer.test.js#L37 add something this:

```js
    t.equal(result, sign(input, Buffer.from(secret)))
    t.equal(result, sign(input, [Buffer.from(secret)]))
```

File: types/plugin.d.ts [hunk 2]
```diff
@@ -128,14 +128,14 @@ declare namespace fastifyCookie {
...
     hook?: HookType | false;
     parseOptions?: fastifyCookie.CookieSerializeOptions;
   }
 
-  export type Sign = (value: string, secret: string, algorithm?: string) => string;
-  export type Unsign = (input: string, secret: string, algorithm?: string) => UnsignResult;
-  export type SignerFactory = (secrets: string | Array<string>, algorithm?: string) => SignerBase;
+  export type Sign = (value: string, secret: string | Buffer, algorithm?: string) => string;
+  export type Unsign = (input: string, secret: string | Buffer, algorithm?: string) => UnsignResult;
+  export type SignerFactory = (secrets: string | Array<string> | Buffer | Array<Buffer>, algorithm?: string) => SignerBase;
```
Comment (Uzlopak) [inline types/plugin.d.ts:138]: I guess we can use only one notation through out the types: 

```suggestion
  export type SignerFactory = (secrets: string | string[] | Buffer | Buffer[], algorithm?: string) => SignerBase;
```

**Review by Uzlopak — APPROVED** [review]

**Review by Eomm — APPROVED** [review]

**Pull request merged**